### ISSUE TITLE
tflint throws a deprecated interpolation warning when terraform requires it

### ISSUE THREAD

**Issue by nirojan** [terraform-linters/tflint#1473]
## Introduction

Tflint warns about using deprecated terraform interpolation when Terraform suggests that I do use the old interpolation when using a variable as key in locals.

```hcl
locals {
  environment_settings = {
    "${var.env}" = {
      a = "b"
      c = "d"
    }
  }
}
```

## Expected Behavior

tflint doesn't warn about old interpolation behaviour in this case as that is what Terraform suggests I do, or it will error.

## Actual behavior

Tflint warns interpolation that is required by Terraform.

```console
❯ tflint
1 issue(s) found:

Warning: Interpolation-only expressions are deprecated in Terraform v0.12.14 (terraform_deprecated_interpolation)

  on locals.tf line 3:
   3:     "${var.env}" = {

Reference: https://github.com/terraform-linters/tflint/blob/v0.39.2/docs/rules/terraform_deprecated_interpolation.md
```

## Step to Reproduce

1. Create locals.tf with following content
```hcl
locals {
  environment_settings = {
    "${var.env}" = {
      a = "b"
      c = "d"
    }
  }
}
```
2. Create vars.tf file with following
```hcl
variable "env" {
  default     = "dev"
}
```
3. Run tflint

## Additional Context

```console
❯ terraform --version
Terraform v1.2.6
on darwin_amd64
❯ tflint --version
TFLint version 0.39.2
+ ruleset.aws (0.15.0)
```

Terraform error if I make the change suggested by tflint

```console
❯ terraform validate
╷
│ Error: Ambiguous attribute key
│
│   on locals.tf line 3, in locals:
│    3:     var.env = {
│
│ If this expression is intended to be a reference, wrap it in parentheses. If it's instead intended as a literal name containing periods, wrap
│ it in quotes to create a string literal.
```

**Comment by bendrucker** [comment]
Duplicate of #1462

**Comment by kawashinji** [comment]
I resolved as below.

```terraform
locals {
  environment_settings = {
    (var.env) = {
      a = "b"
      c = "d"
    }
  }
}
```

See: https://www.terraform.io/language/expressions/types#maps-objects
